R/tmb-methods.R

```diff
@@ -80,6 +82,10 @@ predict.mmrm_tmb <- function(object,
   assert_flag(conditional)
   interval <- match.arg(interval)
   formula_parts <- object$formula_parts
+  # Make sure we have at least a completely NA filled response variable in newdata.
+  if (isFALSE(formula_parts$response_var %in% names(newdata))) {
```

Comment:
whether `response_var` can be a chracter vector? if it is a vector, assiging values may not work (and initially I think adding these variables should already work to predict the prediction intervals)

Reply:
since this is adding much complexity now, I suggest we do the following:

1. add an argument "conditional" to indicate that the prediction is conditional on the current observations
2. in the prediction, if not "conditional" we use `delete.response` to remove the response from terms, and use that terms to re-construct the frame
3. if "conditional" we then create the `h_mmrm_tmb` object

now we don't need to try if the model frame can be construted

Reply:
@clarkliming please have a look at my latest commit which accounts also for this situation. I think it works ok.

Reply:
indeed it solves most of the issue but on rare cases like

```{r}
a = function(){parent.frame()$FEV1}
mmrm(a() ~ AVISIT + ar1(AVISIT|USUBJID),data = fev_data)
```

then the obtained response variable is length 0; this could still lead to issues if FEV1 is not available in the data. but we can proceed with this solution


note: glm works in this scenario because it totally removes the response

```{r}
fit <- glm(a() ~ AVISIT, data = fev_data)

fev2 <- fev_data
fev2$FEV1 <- NULL
predict(fit, newdata = fev2)
```